### PR TITLE
Feature/storage/blob tags

### PR DESCRIPTION
**Pull request by seanmcc-msft**
Note that this doesn't include the new Tags SAS parameter.

### CODE REVIEW

**Comment by seanmcc-msft** [comment]
I think I forgot to implement and test BaseBlobClient.Copy() with Tags.

File: sdk/storage/Azure.Storage.Blobs/src/BlobServiceClient.cs [hunk 1]
```diff
@@ -1323,5 +1324,103 @@ internal BlobServiceClient(Uri serviceUri, HttpPipelinePolicy authentication, Bl
...
+                    $"{nameof(expression)}: {expression}");
+
+                try
+                {
+                    return await BlobRestClient.Service.FilterBlobsAsync(
+                        clientDiagnostics: ClientDiagnostics,
+                        pipeline: Pipeline,
+                        resourceUri: Uri,
+                        version: Version.ToVersionString(),
+                        //where: WebUtility.UrlEncode(expression),
```
Comment (kasobol-msft) [inline sdk/storage/Azure.Storage.Blobs/src/BlobServiceClient.cs:1405]: why is this commented out?

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlobServiceClient.cs:1405]: An oversight, I need to remove it.

Comment (seanmcc-msft) [reply sdk/storage/Azure.Storage.Blobs/src/BlobServiceClient.cs]: Fixed.

**Comment by seanmcc-msft** [comment]
Consolidate UploadBlobOptions and UploadBlockBlobOptions.  UploadLoadBlockBlobOptions isn't necessary, internally BlockBlob.Upload() uses the PartitionedUploader, which takes StorageTranferOptions as a constructor parameters.

**Comment by seanmcc-msft** [comment]
API view: https://apiview.dev/Assemblies/Review/ba1ec17d4c534ae8923719deffeb941b?diffRevisionId=991f312aa09349ae82b8252fe2ba9ce7

Make sure to address all these comments in the PR.

**Comment by seanmcc-msft** [comment]
BlobProperties needs Tag Count.

**Comment by seanmcc-msft** [comment]
Rename FindBlobs -> FindBlobsByTag

**Review by kasobol-msft — APPROVED** [review]

**Pull request merged**